api/openapi-spec/filtered.swagger.json

```diff
@@ -0,0 +1,6742 @@
+{
```

Comment:
What is this file used to do?

Reply:
Hi Derek! This `api/openapi-spec/filtered.swagger.json` is a copy of `api/openapi-spec/swagger.json`. After I upgraded the kube-openapi version, the swagger.json output file was including a new 
 "default" field with a value of an empty string or empty object causing the error that I mentioned in the PR description. To avoid that error, I filtered out those "default" fields from the original swagger.json output and the result of that is a new file, filtered.swagger.json (https://github.com/argoproj/argo-events/blob/72bdef995b759db1441856aa458b13b041086778/hack/update-swagger.sh#L30) that does no longer have those "default" fields.
I made the assumption that filtering out those "default" fields was okay because I noticed it had been previously done for Argo Workflows as well (https://github.com/argoproj/argo-workflows/blob/v3.0.7/hack/swagger/kubeifyswagger.go#L45-L53) @whynowy

Reply:
I see, thanks!

Could you:

1. Only keep the filtered one and name it `swagger.json`, put it at its original directory;
2. Merge `kubeifyswagger` into https://github.com/argoproj/argo-events/blob/master/hack/gen-openapi-spec/main.go?

Reply:
Ok! I moved my kubeifyswagger logic into  https://github.com/argoproj/argo-events/blob/master/hack/gen-openapi-spec/main.go.  Let me know if those changes work for you or if you'd like me to make other adjustments @whynowy